auto-cite/plugins/orcid.py

```diff
@@ -12,14 +13,34 @@ def main(data):
     # list of sources to return
     all_sources = []
 
+    # error exit flag
+    will_exit = False
+
     for index, entry in enumerate(data):
         # show progress
-        log(f"Orcid {index + 1} of {len(data)} - {entry.get('orcid', '-')}", 3, "cyan")
+        log(
+            f"ORCID {index + 1} of {len(data)} - {entry.get('orcid', '[no ORCID]')}",
+            3,
+            "cyan",
+        )
+
+        # get id
+        id = entry.get("orcid")
+        if not id:
+            log('No "orcid" key', 3, "red")
+            will_exit = True
+            continue
 
         # query api to get dois from orcid
-        url = endpoint.replace("$ORCID", entry.get("orcid", "-"))
-        request = Request(url=url, headers=headers)
-        response = json.loads(urlopen(request).read())
+        try:
+            url = endpoint.replace("$ORCID", id)
+            request = Request(url=url, headers=headers)
+            response = json.loads(urlopen(request).read())
+        # if problem with orcid lookup
+        except Exception as message:
```

Comment:
Nit: The convention is `except Exception as e` (or better, `except <MoreSpecificExceptionClass> as e`), since message implies that the variable is a str when it's actually an Exception object. (Also the object has a `.message` attribute and `message.message` is a little weird

That's more for future reference though, this code works fine

Reply:
Will fix this here and anywhere else it occurs on tuesday before merge